fplus-lib/src/core/application/file.rs

```diff
@@ -245,6 +245,7 @@ pub enum AppState {
     AdditionalInfoRequired,
     AdditionalInfoSubmitted,
     Submitted,
+    KYCRequested,
```

Comment:
In `fplus-lib/src/core/mod.rs`, line 3458, in `decline_application`, and in line 3522 (requesting additional info) there's a check that requires state to be `Submitted` or lower to decline the application. I think it's perfectly reasonable to allow Allocator to decline an application if it's in `KYCRequested`, so let's move it above `Submitted`.